### PR TITLE
Rewrite the Grafeas reference implementation to use grpc

### PR DESCRIPTION
**Pull request by R2wenD2**
Clients still need to be updated.

### CODE REVIEW

**Comment by R2wenD2** [comment]
@vbatts Can you also TAL. When I update the clients (later this week), I'll include a main in them that can convert from json to proto so that testing from the command line is a bit easier.

**Comment by R2wenD2** [comment]
@furuholm this will also impact some of the work you have in flight.

**Comment by furuholm** [comment]
make fails for me with the error

```
vet: ./test/go/grafeas.go: ./test/go/grafeas.go:3:1: expected 'IDENT', found 'import'
vet: no files checked
make: *** [vet] Error 1
```

File: Makefile [hunk 1]
```diff
@@ -29,8 +29,8 @@ v1alpha1/proto/grafeas.pb.go: .install.protoc-gen-go v1alpha1/proto/grafeas.prot
 		-I ./ \
 		-I vendor/github.com/grpc-ecosystem/grpc-gateway/third_party/googleapis \
 		-I vendor/github.com/googleapis/googleapis \
-		--go_out=./ \
-		v1alpha1/proto/grafeas.proto
+	    v1alpha1/proto/grafeas.proto \
```
Comment (vbatts) [inline Makefile]: Are these lines meaning to be switched around?

File: samples/server/go-server/api/server/name/name.go [hunk 2]
```diff
@@ -131,56 +130,56 @@ func ParseResourceKindAndResource(name string) (kind ResourceKind, pID, eID stri
...
+			noteNameFormat, operationNameFormat), parent)
 	}
 	return Unknown, "", err
 }
 
 // ParseOccurrence takes a stringly typed name of the form:
 //   projects/{project_id}/occurrences/{occurrence_id}
 // validates its form and returns either an error or the project-/occurrence-ids.
-func ParseOccurrence(name string) (string, string, *errors.AppError) {
+func ParseOccurrence(name string) (string, string, error) {
```
Comment (vbatts) [inline samples/server/go-server/api/server/name/name.go]: Glad we're going to just error type. I had meant to do that.

**Comment by vbatts** [comment]
I love seeing this much deletion.

File: samples/server/go-server/api/server/main/main.go [hunk 3]
```diff
@@ -16,17 +16,33 @@ package main
...
+	port = flag.Int("port", 10000, "The server port")
 )
 
 func main() {
-	log.Printf("Server started on port 8080")
-	s := storage.NewMemStore()
-	router := server.NewRouter(v1alpha1.Grafeas{S: s})
-	log.Fatal(http.ListenAndServe(":8080", router))
+	flag.Parse()
+	log.Printf("Server started on port %v", *port)
```
Comment (vtsao) [inline samples/server/go-server/api/server/main/main.go]: %d

File: samples/server/go-server/api/server/name/name.go [hunk 4]
```diff
@@ -56,9 +56,8 @@ var (
 	noteNameFormat       = FormatNote("{provider_project_id}", "{note_id}")
 )
 
-func invalidArg(pattern, got string) *errors.AppError {
-	return &errors.AppError{Err: fmt.Sprintf("expected name to be of form %q, input was %v", pattern, got),
-		StatusCode: http.StatusBadRequest}
+func invalidArg(pattern, got string) error {
+	return status.Error(codes.InvalidArgument, fmt.Sprintf("expected name to be of form %q, input was %v", pattern, got))
```
Comment (vtsao) [inline samples/server/go-server/api/server/name/name.go]: %q

File: samples/server/go-server/api/server/name/name_test.go [hunk 5]
```diff
@@ -99,7 +112,7 @@ func TestParseNoteValidation(t *testing.T) {
 	}
 
 	for _, test := range badNoteNames {
-		if p1, p2, err := ParseNote(test); err.Err == "" {
+		if p1, p2, err := ParseNote(test); err == nil {
 			t.Errorf("ParseNote %v; wanted error, got (%v, %v)",
```
Comment (vtsao) [inline samples/server/go-server/api/server/name/name_test.go]: Got before want.

File: samples/server/go-server/api/server/name/name_test.go [hunk 6]
```diff
@@ -198,23 +211,45 @@ func TestParseResourceKindAndProjectPath(t *testing.T) {
...
+		"projects/foo/baz",
+		// Too many parts
+		"projects/foo/baz/asdf",
+		// Empty part
+		"projects//",
+	}
+
+	for _, test := range badProjectNames {
+		if p1, err := ParseProject(test); err == nil {
+			t.Errorf("ParseProject %v; wanted error, got %v",
```
Comment (vtsao) [inline samples/server/go-server/api/server/name/name_test.go]: got before want

File: Makefile (hunk 1 above)
Comment (R2wenD2) [reply Makefile]: Yes - following the grpc instructions from https://grpc.io/docs/quickstart/go.html

File: samples/server/go-server/api/server/main/main.go (hunk 3 above)
Comment (R2wenD2) [reply samples/server/go-server/api/server/main/main.go]: done

File: samples/server/go-server/api/server/name/name.go (hunk 4 above)
Comment (R2wenD2) [reply samples/server/go-server/api/server/name/name.go]: done

File: samples/server/go-server/api/server/name/name_test.go (hunk 5 above)
Comment (R2wenD2) [reply samples/server/go-server/api/server/name/name_test.go]: done

File: samples/server/go-server/api/server/name/name_test.go (hunk 6 above)
Comment (R2wenD2) [reply samples/server/go-server/api/server/name/name_test.go]: done

**Comment by R2wenD2** [comment]
@furuholm I think your make is failing on the client libs? I need to rework those - that's next on my list, as well as using grpc gateway so that we can still use json/REST against the API. https://github.com/grpc-ecosystem/grpc-gateway

**Pull request merged**